openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java

```diff
@@ -0,0 +1,279 @@
+    this.env = env;
+  }
+
+  @Autowired
+  public void setContext(ApplicationContext context) {
+    this.context = context;
+  }
+
+  private List<OAuthProvider> buildOpenIdProviders() {
+    if (!this.openBASConfig.isAuthOpenidEnable()) {
+      return new ArrayList<>();
+    }
+    try {
+      OAuth2ClientProperties properties = this.context.getBean(OAuth2ClientProperties.class);
+      Map<String, OAuth2ClientProperties.Registration> providers = properties.getRegistration();
+      return providers.entrySet().stream()
+          .map(entry -> {
+            String uri = "/oauth2/authorization/" + entry.getKey();
+            String clientName = env.getProperty("openex.provider." + entry.getKey() + ".login");
+            // In case of missing name configuration, generate a generic name
+            if (clientName == null) {
+              clientName = "Login with " + entry.getKey();
+            }
+            return new OAuthProvider(entry.getKey(), uri, clientName);
+          })
+          .toList();
+    } catch (Exception e) {
+      // No provider defined in the configuration
+      return new ArrayList<>();
+    }
+  }
+
+  private Map<String, Setting> mapOfSettings() {
+    return fromIterable(this.settingRepository.findAll()).stream().collect(
+        Collectors.toMap(Setting::getKey, Function.identity()));
+  }
+
+  private String getValueFromMapOfSettings(@NotBlank final String key) {
+    return Optional.ofNullable(mapOfSettings().get(key)).map(Setting::getValue).orElse(null);
+  }
+
+  private List<OAuthProvider> buildSaml2Providers() {
+    if (!this.openBASConfig.isAuthSaml2Enable()) {
+      return new ArrayList<>();
+    }
+    try {
+      Saml2RelyingPartyProperties properties = this.context.getBean(Saml2RelyingPartyProperties.class);
+      Map<String, Saml2RelyingPartyProperties.Registration> providers = properties.getRegistration();
+      return providers.entrySet().stream()
+          .map(entry -> {
+            String uri = "/saml2/authenticate/" + entry.getKey();
+            String clientName = env.getProperty("openex.provider." + entry.getKey() + ".login");
+            // In case of missing name configuration, generate a generic name
+            if (clientName == null) {
+              clientName = "Login with " + entry.getKey();
+            }
+            return new OAuthProvider(entry.getKey(), uri, clientName);
+          })
+          .toList();
+    } catch (Exception e) {
+      // No provider defined in the configuration
+      return new ArrayList<>();
+    }
+  }
+
+  private Setting resolveFromMap(Map<String, Setting> dbSettings, String themeKey, String value) {
+    Optional<Setting> optionalSetting = ofNullable(dbSettings.get(themeKey));
+    if (optionalSetting.isPresent()) {
+      Setting updateSetting = optionalSetting.get();
+      updateSetting.setValue(value);
+      return updateSetting;
+    }
+    return new Setting(themeKey, value);
+  }
+
+  @GetMapping("/api/platform/settings")
```

Comment:
It's better to have one endpoint to handle PlatformSettings.
And not both: /api/platform/settings & /api/settings

We need to finish refactoring on the front-end side (I know, I changed my mind :D )
And keep the /api/settings uri to be consistent with the other one: /api/settings/theme/*